src/core/tsi/ssl_transport_security.h

```diff
@@ -223,6 +223,10 @@ tsi_result tsi_ssl_client_handshaker_factory_create_handshaker(
     const char* server_name_indication, size_t network_bio_buf_size,
     size_t ssl_bio_buf_size, tsi_handshaker** handshaker);
 
+// Increments reference count of the client handshaker factory.
+tsi_ssl_client_handshaker_factory* tsi_ssl_client_handshaker_factory_ref(
```

Comment:
Please add unit tests for this function to make sure it plays well with unref-ing.

Reply:
Added to a test in `ssl_transport_security_test.cc`